include/geos/algorithm/distance/DiscreteFrechetDistance.h

```diff
@@ -0,0 +1,179 @@
+ * See the COPYING file for more information.
+ *
+ **********************************************************************
+ *
+ * Last port: algorithm/distance/DiscreteFrechetDistance.java 1.5 (JTS-1.10)
+ *
+ **********************************************************************/
+
+#ifndef GEOS_ALGORITHM_DISTANCE_DISCRETEFRECHETDISTANCE_H
+#define GEOS_ALGORITHM_DISTANCE_DISCRETEFRECHETDISTANCE_H
+
+#include <geos/export.h>
+#include <geos/algorithm/distance/PointPairDistance.h> // for composition
+#include <geos/algorithm/distance/DistanceToPoint.h> // for composition
+#include <geos/util/IllegalArgumentException.h> // for inlines
+#include <geos/geom/Geometry.h> // for inlines
+#include <geos/util/math.h> // for inlines
+#include <geos/geom/CoordinateFilter.h> // for inheritance
+#include <geos/geom/CoordinateSequence.h> // for inheritance
+
+#include <cstddef>
+#include <vector>
+
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable: 4251) // warning C4251: needs to have dll-interface to be used by clients of class
+#endif
+
+namespace geos {
+	namespace algorithm {
+		//class RayCrossingCounter;
+	}
+	namespace geom {
+		class Geometry;
+		class Coordinate;
+		//class CoordinateSequence;
+	}
+	namespace index {
+		namespace intervalrtree {
+			//class SortedPackedIntervalRTree;
+		}
+	}
+}
+
+namespace geos {
+namespace algorithm { // geos::algorithm
+namespace distance { // geos::algorithm::distance
+
+/** \brief
+ * An algorithm for computing a distance metric
+ * which is an approximation to the Frechet Distance
+ * based on a discretization of the input {@link Geometry}.
+ *
+ * The algorithm computes the Frechet distance restricted to discrete points
+ * for one of the geometries.
+ * The points can be either the vertices of the geometries (the default),
+ * or the geometries with line segments densified by a given fraction.
+ * Also determines two points of the Geometries which are separated by the
+ * computed distance.
+ *
+ * This algorithm is an approximation to the standard Frechet distance.
+ * Specifically,
+ * <pre>
+ *    for all geometries a, b:    DHD(a, b) <= HD(a, b)
+ * </pre>
+ * The approximation can be made as close as needed by densifying the
+ * input geometries.
+ * In the limit, this value will approach the true Frechet distance:
+ * <pre>
+ *    DHD(A, B, densifyFactor) -> HD(A, B) as densifyFactor -> 0.0
+ * </pre>
+ * The default approximation is exact or close enough for a large subset of
+ * useful cases.
+ * Examples of these are:
+ *
+ * - computing distance between Linestrings that are roughly parallel to
+ *   each other, and roughly equal in length.  This occurs in matching
+ *   linear networks.
+ * - Testing similarity of geometries.
+ *
+ * An example where the default approximation is not close is:
+ * <pre>
+ *   A = LINESTRING (0 0, 100 0, 10 100, 10 100)
+ *   B = LINESTRING (0 100, 0 10, 80 10)
+ *
+ *   DHD(A, B) = 22.360679774997898
+ *   HD(A, B) ~= 47.8
+ * </pre>
+ */
```

Comment:
Is this documentation correct ? As it looks like just the Hausdorff distance doc with a search & replace